### PR TITLE
Correct user manager setter

### PR DESCRIPTION
**Pull request by suhasj**
@HaoK @divega 
The old PR was not restored, hence this new one. 

Changes
- removed the private method for UpdateUserName since the normalizedUsername is called on UpdateUserAsync method
- Added unit test to check validation happen on set username/ password

### CODE REVIEW

File: test/Shared/UserManagerTestBase.cs [hunk 1]
```diff
@@ -128,15 +128,56 @@ protected RoleManager<TRole> CreateRoleManager(object context = null, IServiceCo
...
+
+        [Fact]
+        public async Task CheckSetEmailValidatesUser()
+        {
+            var manager = CreateManager();
+            manager.Options.User.RequireUniqueEmail = true;
+            manager.UserValidators.Add(new UserValidator<TUser>());
+            var random = new Random();
+            var email = "foo" + random.Next() + "@example.com";
+            var newEmail = "foo" + random.Next() + "@example.com";
```
Comment (HaoK) [inline test/Shared/UserManagerTestBase.cs]: Why risk collisions?  Change the prefix

File: src/Microsoft.AspNet.Identity/UserManager.cs [hunk 2]
```diff
@@ -1998,7 +1993,7 @@ public virtual Task<IList<TUser>> GetUsersInRoleAsync(string roleName)
             var state = Resources.FormatLoggingResultMessageForUser(methodName, await GetUserIdAsync(user));
             return Logger.BeginScope(state);
         }
-            
+
```
Comment (HaoK) [inline src/Microsoft.AspNet.Identity/UserManager.cs]: Empty line added

File: test/Shared/UserManagerTestBase.cs (hunk 1 above)
Comment (suhasj) [inline test/Shared/UserManagerTestBase.cs]: sure

**Comment by HaoK** [comment]
:shipit:

**Comment by suhasj** [comment]
@divega Reminder

**Comment by divega** [comment]
:shipit:

**Pull request closed without merging**